docs/design/data_model.md

```diff
@@ -0,0 +1,412 @@
+| | _RegisterDeleteEntry_ | Entry | Original entry address |
+| | _RegisterLink_ | Link | Action address, from the link's base |
+| | _RegisterDeleteLink_ | Link | Action address, from the deleted link's base |
+| Warrant | _InvalidChainOp_ | Agent | Agent address of the agent who authored invalid data |
+| | _ChainFork_ | Agent | Agent address of the agent who forked their chain |
+
+The transformation from source chain actions to DHT operations serves several purposes:
+
+**Selective distribution:** Different authorities receive only the data they need to perform their role. The entry authority receives the entry content, while the agent activity authority receives only the action without the entry.
+
+**Enabling queries:** DHT operations establish the indices and metadata needed for efficient queries. Links are stored at their base address, updates are indexed at the original entry, and agent activity is indexed by agent public key.
+
+### Chain ops
+
+Each action type is mapped to a specific set of DHT operations. This happens after committing to a source chain but the mapping can be done at any later point and still be valid. The following table shows which operations are generated for each action:
+
+| Action Type | Ops Produced |
+|-------------|---------------------|
+| **Dna** | _StoreRecord_, _RegisterAgentActivity_ |
+| **AgentValidationPkg** | _StoreRecord_, _RegisterAgentActivity_ |
+| **InitZomesComplete** | _StoreRecord_, _RegisterAgentActivity_ |
+| **Create** | _StoreRecord_, _RegisterAgentActivity_, _StoreEntry_ |
+| **Update** | _StoreRecord_, _RegisterAgentActivity_, _StoreEntry_, _RegisterUpdateRecord_, _RegisterUpdateEntry_ |
+| **Delete** | _StoreRecord_, _RegisterAgentActivity_, _RegisterDeleteRecord_, _RegisterDeleteEntry_ |
+| **CreateLink** | _StoreRecord_, _RegisterAgentActivity_, _RegisterLink_ |
+| **DeleteLink** | _StoreRecord_, _RegisterAgentActivity_, _RegisterDeleteLink_ |
+
+Note that every action produces a:
+- _StoreRecord_ op meaning that every action is content addressible by its own action hash, and
+- _RegisterAgentActivity_, which meaning that every action is sent to the author's authorities.
+
+Each chain op contains the source action, and a signature of that action by its author. Note that the entry data is not signed, but entries are always referenced by their hash by an action so the signature of the action covers the entry data as long as the entry hash is checked.
+
+Whether an entry is included in an op depends on the type of action, content of the action and the type of op. As described for actions and records above, some actions do not have an associated entry and entries may also be hidden. The _RegisterAgentActivity_ op never carries an entry, and neither do the _RegisterDeleteRecord_ or _RegisterDeleteEntry_ types.
+
+### Chain Operations
+
+There are nine types of chain operations, each serving a specific purpose in the distributed data model:
+
+#### RegisterAgentActivity
```

Comment:
I suggest to drop all the `Register`s from the op identifiers and rename `RegisterLink` to `CreateLink`.